OOModel/src/expressions/MetaCallExpression.cpp

```diff
@@ -62,4 +69,61 @@ MetaDefinition* MetaCallExpression::metaDefinition()
 	return ret;
 }
 
+void MetaCallExpression::bindMetaCalls(Model::Node* n, MetaBinding* binding)
+{
+	if (auto metaCall = DCast<MetaCallExpression>(n))
+		if (auto callee = DCast<ReferenceExpression>(metaCall->callee()))
+			for (auto i = 0; i < binding->mappings()->size(); i++)
+				if (callee->name() == binding->mappings()->at(i)->name())
+					callee->setName(binding->mappings()->at(i)->value()->name());
+
+	for (auto child : n->children()) bindMetaCalls(child, binding);
+}
+
+Declaration* MetaCallExpression::generate()
+{
+	// return cached generated tree if exists
+	if (generated()) return generated();
```

Comment:
At some point we have to think about invalidating this cache. Not in this PR, but you should put it on your todo list.